_layouts/post.html

```diff
@@ -16,9 +16,21 @@
   {% include header.html %}
 
   <main>
-    <div class="post">
-      <h1>{{ page.title }}</h1>
+    <div class="post-header">
+      {% if page.header-image %}
+        <img class="post-header-image" src="{{ page.header-image | relative_url }}">
+      {% elsif page.thumbnail %}
```

Comment:
Not a huge fan of this alternative - the thumbnails are meant to be smaller images, so adding them as a banner image will stretch them and make them look weird in a lot of cases. I'd just keep the header image property, and choose not to display the thumbnail on the post at all.